find_package(storage_client REQUIRED) fails from CMake

**What component of `google-cloud-cpp` is this related to?**
GCS

**Describe the bug**
Cannot use GCS library from CMake script as `find_package(storage_client REQUIRED)` fails.

**To Reproduce**
Steps to reproduce the behavior:
1. Build google-cloud-cpp using CMake with following parameters:
```
BUILD_TESTING=OFF
GOOGLE_CLOUD_CPP_ENABLE=storage
GOOGLE_CLOUD_CPP_ENABLE_GRPC=OFF
GOOGLE_CLOUD_CPP_ENABLE_BIGQUERY=OFF
GOOGLE_CLOUD_CPP_ENABLE_BIGTABLE=OFF
GOOGLE_CLOUD_CPP_ENABLE_SPANNER=OFF
GOOGLE_CLOUD_CPP_ENABLE_FIRESTORE=OFF
GOOGLE_CLOUD_CPP_ENABLE_WERROR=OFF
```
2. install the google-cloud-cpp using CMake
3. In consumer project's CMake script write:
```cmake
find_package(storage_client REQUIRED)
```
4. See error:
```
1>CMake Error at C:/Program Files/CMake/share/cmake-3.19/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
1>  By not providing "Findgoogleapis.cmake" in CMAKE_MODULE_PATH this project
1>  has asked CMake to find a package configuration file provided by
1>  "googleapis", but CMake did not find one.
1>
1>  Could not find a package configuration file provided by "googleapis" with
1>  any of the following names:
1>
1>    googleapisConfig.cmake
1>    googleapis-config.cmake
1>
1>  Add the installation prefix of "googleapis" to CMAKE_PREFIX_PATH or set
1>  "googleapis_DIR" to a directory containing one of the above files.  If
1>  "googleapis" provides a separate development package or SDK, be sure it has
1>  been installed.
1>Call Stack (most recent call first):
1>  xxx/cmake/storage_client/storage_client-config.cmake:17 (find_dependency)
1>  zzz/CMakeLists.txt:65 (find_package)
1>
1>
1>-- Configuring incomplete, errors occurred!
```

**Expected behavior**
`find_package(storage_client REQUIRED)` CMake code should work.

**Operating system:**
Windows 10.

**What compiler and version are you using?**
MS VS 2019 16.8.4.

**What version of `google-cloud-cpp` are you using?**
1.24.0.

**Additional context**
Previous version 1.23.0 worked ok.
CMake version 3.19.0

--- a/google/cloud/storage/legacy.cmake.in
+++ b/google/cloud/storage/legacy.cmake.in
@@ -14,7 +14,6 @@
 
 include(CMakeFindDependencyMacro)
 find_dependency(google_cloud_cpp_storage)
-find_dependency(googleapis)
 
 foreach (target storage_client)
     if (TARGET ${target})

